libs/utils/WordPressUtils/src/main/java/org/wordpress/android/util/ImageUtils.java

```diff
@@ -534,44 +486,153 @@ public static String createResizedImageWithMaxWidth(Context context, String path
 
         float scaleBy = Math.min(scaleWidth, scaleHeight);
 
-        // Resize the bitmap to exact size
         Matrix matrix = new Matrix();
         matrix.postScale(scaleBy, scaleBy);
 
         // apply rotation
-        if (rotation != 0) {
-            matrix.setRotate(rotation);
+        if (orientation != 0) {
+            matrix.setRotate(orientation);
         }
 
         Bitmap.CompressFormat fmt;
-        if (fileExtension != null && fileExtension.equalsIgnoreCase("png")) {
+        if (fileExtension.equalsIgnoreCase("png")) {
             fmt = Bitmap.CompressFormat.PNG;
         } else {
             fmt = Bitmap.CompressFormat.JPEG;
         }
 
         final Bitmap bmpRotated;
         try {
-            bmpRotated = Bitmap.createBitmap(bmpResized, 0, 0, bmpResized.getWidth(), bmpResized.getHeight(), matrix,
-                    true);
+            bmpRotated = Bitmap.createBitmap(bmpResized, 0, 0, bmpResized.getWidth(), bmpResized.getHeight(), matrix, true);
+            bmpResized.recycle();
         } catch (OutOfMemoryError e) {
-            AppLog.e(AppLog.T.UTILS, "OutOfMemoryError Error in setting image: " + e);
-            return null;
+            AppLog.e(AppLog.T.UTILS, "OutOfMemoryError while creating the resized bitmap", e);
+            throw e;
         } catch (NullPointerException e) {
             // See: https://github.com/wordpress-mobile/WordPress-Android/issues/1844
-            AppLog.e(AppLog.T.UTILS, "Bitmap.createBitmap has thrown a NPE internally. This should never happen: " + e);
-            return null;
+            AppLog.e(AppLog.T.UTILS, "Bitmap.createBitmap has thrown a NPE internally. This should never happen!", e);
+            throw e;
         }
 
         if (bmpRotated == null) {
             // Fix an issue where bmpRotated is null even if the documentation doesn't say Bitmap.createBitmap can return null.
+            AppLog.e(AppLog.T.UTILS, "bmpRotated is null even if the documentation doesn't say Bitmap.createBitmap can return null.");
             // See: https://github.com/wordpress-mobile/WordPress-Android/issues/1848
-            return null;
+            throw new IOException("bmpRotated is null even if the documentation doesn't say Bitmap.createBitmap can return null.");
         }
 
-        bmpRotated.compress(fmt, 100, stream);
-        bmpResized.recycle();
+        boolean result = bmpRotated.compress(fmt, quality, outStream);
         bmpRotated.recycle();
```

Comment:
I moved the call to `recycle` on bmpResized immediately before instantiating the rotated bitmap, so the system knows it can reuse the memory allocated for bmpResized if needed. I think the compiler already does this kind of things by itself, by marking that variable `bmpResized` as unused after that line of code, and it recycles the memory.